code/game/machinery/doors/door.dm

```diff
@@ -77,9 +77,14 @@
 		var/mob/living/M = AM
 		if(world.time - M.last_bumped <= 10) return	//Can bump-open one airlock per second. This is to prevent shock spam.
 		M.last_bumped = world.time
-		if(!M.restrained() && M.mob_size > MOB_SIZE_SMALL)
-			bumpopen(M)
-		return
+		if(!M.restrained())
+			if(istype(AM, /mob/living/simple_animal))
+				var/mob/living/simple_animal/A = AM
+				if(A.can_collar)

```

Comment:
Tbh I wanted to check if they collared, but I don't think there is anything specific in place for that at this time; nor did I wish to figure out how to write that at the time. As far as access goes: If the ID inserted in the collar doesn't have appropriate access to the door attempting to be bumped open, then the door will deny entry. I figured it was handled somewhere down the line.. I didn't trace down where in the code, unfortunately.